webapp/channels/src/components/create_team/components/team_url/team_url.test.tsx

```diff
@@ -65,15 +50,28 @@ describe('/components/create_team/components/display_name', () => {
         const actions = {...defaultProps.actions, checkIfTeamExists};
         const props = {...defaultProps, actions};
 
-        const wrapper = mountWithIntl(
+        renderWithContext(
             <TeamUrl {...props}/>,
         );
 
-        await (wrapper.instance() as unknown as TeamUrl).submitNext({preventDefault: jest.fn()} as unknown as React.MouseEvent<Button, MouseEvent>);
+        screen.getByText('Finish').click();
+
+        // Wait for the check to see if the team already exists which should fail
+        await Promise.resolve();
+
+        expect(screen.getByText('This URL is taken or unavailable. Please try another.')).toBeInTheDocument();
+
         expect(actions.checkIfTeamExists).toHaveBeenCalledTimes(1);
         expect(actions.createTeam).not.toHaveBeenCalled();
 
-        await (wrapper.instance() as unknown as TeamUrl).submitNext({preventDefault: jest.fn()} as unknown as React.MouseEvent<Button, MouseEvent>);
+        expect(screen.getByText('This URL is taken or unavailable. Please try another.')).toBeInTheDocument();
+
+        screen.getByText('Finish').click();
+
+        // Wait for the check to see if the team already exists and then to create the team
+        await Promise.resolve();
+        await Promise.resolve();
```

Comment:
Why two awaits?

Reply:
The submit handler awaits on two different actions, so we have to await twice to let those proceed. I wasn't able to find a way to await every outstanding promise, but I'll take another look.

Also, the comment was meant to explain that, but I guess it isn't clear enough 😛